code/core/src/components/components/tooltip/WithTooltip.tsx

```diff
@@ -177,7 +177,12 @@ const WithToolTipState = ({
 
   useEffect(() => {
     const hide = () => onVisibilityChange(false);
-    document.addEventListener('keydown', hide, false);
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        hide();
+      }
+    };
+    document.addEventListener('keydown', handleKeyDown, false);
```

Comment:
@yannbf not a blocker to this PR, but I want to point out that this type of behavior is a good candidate for writing tests for.

- open model
- keypress "A"
- assert modal does not close
- keypress "Esc"
- assert modal does indeed close